### PR TITLE
Fix: Previous course user name and roles appear in the modal

### PR DESCRIPTION
**Pull request by AlreadyBored**
**🟢 Add `deploy` label if you want to deploy this Pull Request to staging environment**

#### 🧑‍⚖️ Pull Request Naming Convention

- Title should follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- Do not put issue id in title
- Do not put WIP in title. Use [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) functionality
- Consider to add `area:*` label(s)

* [x] I followed naming convention rules

---

#### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related issue link

https://github.com/rolling-scopes/rsschool-app/issues/2035

#### 💡 Background and solution

Fixed bug when previous user's github id and roles appear in course users modal after click on **Edit** or **Add User button**

#### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Database migration is added or not needed
- [x] Documentation is updated/provided or not needed
- [x] Changes are tested locally

### CODE REVIEW

File: client/src/components/Forms/ModalForm.tsx [hunk 1]
```diff
@@ -22,7 +22,12 @@ export function ModalForm(props: Props) {
   if (props.data == null) {
     return null;
   }
-  const initialValues = props.getInitialValues ? props.getInitialValues?.(props.data) : props.data;
+
+  const formValues = props.getInitialValues ? props.getInitialValues?.(props.data) : props.data;
+
+  form.resetFields();
```
Comment (apalchys) [inline client/src/components/Forms/ModalForm.tsx:28]: i don't think it's ok to reset on every component render globally

Comment (AlreadyBored) [reply client/src/components/Forms/ModalForm.tsx]: Yep, we should do it only on new data come, fixed

**Review by apalchys — APPROVED** [review]

File: client/src/components/Forms/ModalForm.tsx [hunk 2]
```diff
@@ -19,10 +19,20 @@ export function ModalForm(props: Props) {
...
+  }, [props.data]);
+
   if (props.data == null) {
     return null;
   }
-  const initialValues = props.getInitialValues ? props.getInitialValues?.(props.data) : props.data;
+
+  const formValues = props.getInitialValues ? props.getInitialValues?.(props.data) : props.data;
+
+  form.setFieldsValue(formValues);
```
Comment (aaliakseyenka) [inline client/src/components/Forms/ModalForm.tsx:34]: wondering: why to update form manually on each render? what's wrong with initial values? looks kind of buggy

Comment (AlreadyBored) [reply client/src/components/Forms/ModalForm.tsx:34]: Oh, didn't noticed your comment. I think it's related to Antd modal/form weird persistance somehow, it sets proper values from second attempt. We wanted to get rid of this component soon anyway, so I didn't dived too deep into it :)

**Pull request merged**